hs_core/tasks.py

```diff
@@ -250,3 +250,15 @@ def create_bag_by_irods(resource_id):
     else:
         logger.error('Resource does not exist.')
         return False
+
+
+@shared_task
+def check_irods_files():
+    """
+    Check every resource for iRODS file problems.
+
+    This logs ERROR messages for every problem it finds
+    """
+    for r in BaseResource.objects.all():
+        r.check_irods_files(log_errors=True, echo_errors=False,
+                            stop_on_error=False, return_errors=False)

```

Comment:
This check_irods_files() should not be added here as a shared_task which is meant to be executed/called from Django and executed asynchronously. Since you have already added a management command for check irods files, there is no point to add it as a celery task here, since that management command can be hooked up in Jenkins if there is a need to run it as a cron job to run in a regular interval. You can remove this.